http/httputils.h

```diff
@@ -0,0 +1,159 @@
+ *
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ */
+
+/**
+ * @file httputils.h
+ * @brief HTTP(S) utility API built on top of libcurl for the Greenbone framework.
+ *
+ * This module provides a high-level wrapper around libcurl to simplify the process
+ * of performing HTTP and HTTPS requests in both synchronous and asynchronous (multi)
+ * modes. It offers abstractions for:
+ *
+ * - Sending requests with various HTTP methods (GET, POST, PUT, DELETE, etc.)
+ * - Managing custom request headers
+ * - SSL/TLS authentication with CA, client certificates, and private keys
+ * - Accumulating and handling response data via callback streams
+ * - Managing and cleaning up single and multi-handle CURL resources
+ *
+ * Core data structures:
+ * - `gvm_http_t`: encapsulates a single HTTP request configuration and state
+ * - `gvm_http_response_t`: represents the HTTP response including status, data, and associated request
+ * - `gvm_http_headers_t`: stores custom headers for use in requests
+ * - `gvm_http_response_stream_t`: used internally for accumulating response data during transfers
+ * - `gvm_http_multi_t`: manages multiple concurrent transfers using libcurl's multi interface
+ */
+
+#ifndef HTTPUTILS_H
+#define HTTPUTILS_H
+
+#include <curl/curl.h>
+#include <curl/easy.h>
+#include <curl/multi.h>
+#include <glib.h>
+
+/**
+ * @brief Request methods
+ */
+typedef enum {
+  GET,
+  POST,
+  PUT,
+  DELETE,
+  HEAD
+} gvm_http_method_t;
+
+typedef enum {
+  GVM_HTTP_OK,
+  GVM_HTTP_MULTI_BAD_HANDLE,
+  GVM_HTTP_MULTI_FAILED,
+  GVM_HTTP_MULTI_UNKNOWN_ERROR
+} gvm_http_multi_result_t;
+
+typedef struct gvm_http_headers {
+
+  struct curl_slist *custom_headers;
+
+} gvm_http_headers_t;
+
+/**
+ * @brief Wraps a CURLM * handler and the custom headers.
+ */
+typedef struct gvm_http_multi
+{
+  void *handler;                ///< Opaque pointer to the internal CURLM handle.
+
+  gvm_http_headers_t *headers; ///< The wrapped headers type.
+
+} gvm_http_multi_t;
+
+/**
+ * @brief Defines a struct for storing the response and http multi-handler.
+ */
+typedef struct gvm_http_response_stream
+{
+  gchar *data; ///< Pointer to the accumulated response data buffer.
+
+  size_t length; ///< Length of the response data buffer.
+
+  gvm_http_multi_t *multi_handler; ///< Pointer to the associated http
+                                   ///< multi-handle and headers.
+} gvm_http_response_stream;
+
+typedef struct gvm_http_response_stream *gvm_http_response_stream_t;
+
+typedef struct {
+
+  CURL *handler;
+
+} gvm_http_t;
+
+/**
+ * @brief Represents the result of a http request.
+ */
+typedef struct {
+  gchar *data; ///< The actual response content as a string.
+
+  gsize size; ///< Size of the response content.
+
+  glong http_status; ///< HTTP status code returned by the server.
+
+  gvm_http_t *http; ///< The HTTP request (easy handle wrapper).
+} gvm_http_response_t;
+
+gvm_http_t *
+gvm_http_new (CURL *curl_handler);
```

Comment:
Is there any reason to export this? It's the only thing I see that needs the curl includes in this header.